src/libraries/System.Text.Json/System.Text.Json.SourceGeneration/ReflectionUtils/MetadataLoadContext.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.CodeAnalysis;
+
+namespace System.Reflection
+{
+    public class MetadataLoadContext
+    {
+        private readonly Dictionary<string, IAssemblySymbol> _assemblies = new Dictionary<string, IAssemblySymbol>(StringComparer.OrdinalIgnoreCase);
+        private readonly Compilation _compilation;
+
+        public MetadataLoadContext(Compilation compilation)
+        {
+            _compilation = compilation;
+            var assemblies = compilation.References
```

Comment:
Use type name instead of var whenever RHS does not have `new Type(...)` or a cast to the type. See #10 in the coding style guidelines: https://github.com/dotnet/runtime/blob/3c9fcce29d00d0d38148c4113b17aeeaa5cbc970/docs/coding-guidelines/coding-style.md.

Reply:
This applies here and elsewhere